### PR TITLE
feat(api) - Support Uploading Excel Files for RAG

### PR DESCRIPTION
**Pull request by CollectiveUnicorn**
Adds excel (`xlsx`) support to files the endpoints.
Fixes issue with broken file content download endpoint.

Closes: #622
Closes: #585
Closes: #675

### CODE REVIEW

File: src/leapfrogai_api/routers/openai/files.py [hunk 1]
```diff
@@ -126,7 +131,30 @@ async def retrieve_file_content(
...
         crud_file_bucket = CRUDFileBucket(db=session, model=UploadFile)
-        return await crud_file_bucket.download(id_=file_id)
+        file_content = await crud_file_bucket.download(id_=file_id)
+
+        # Get the file object to retrieve the filename
+        crud_file_object = CRUDFileObject(session)
+        file_object = await crud_file_object.get(filters=FilterFileObject(id=file_id))
+
+        if not file_object:
+            raise HTTPException(status_code=404, detail="File not found")
```
[the remark this answers is not in the document]
Comment (CollectiveUnicorn) [reply src/leapfrogai_api/routers/openai/files.py]: Replaced

File: tests/integration/api/test_files.py
[the remark this answers is not in the document]
Comment (CollectiveUnicorn) [reply tests/integration/api/test_files.py:1]: Replaced the hardcoded value with the status code

File: src/leapfrogai_api/backend/rag/document_loader.py [hunk 2]
```diff
@@ -23,8 +24,30 @@
...
+EXTENSION_TO_MIME_TYPE = {
+    ".pdf": "application/pdf",
+    ".txt": "text/plain",
+    ".html": "text/html",
+    ".htm": "text/html",
+    ".csv": "text/csv",
+    ".md": "text/markdown",
+    ".doc": "application/msword",
+    ".docx": "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
+    ".xlsx": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
```
Comment (gphorvath) [inline src/leapfrogai_api/backend/rag/document_loader.py:40]: does this loader support `xls:application/vnd.ms-excel` and `xlsm:application/vnd.ms-excel.sheet.macroEnabled.12`?

**Review by gphorvath — DISMISSED** [review]
Save my one optional comment about expanding support for other excel mime types, this looks good to me!

File: src/leapfrogai_api/backend/rag/document_loader.py (hunk 2 above)
Comment (CollectiveUnicorn) [reply src/leapfrogai_api/backend/rag/document_loader.py:41]: It doesn't support `xlsm:application/vnd.ms-excel.sheet.macroEnabled.12` but it does support `xls:application/vnd.ms-excel`. I've added 'xls:application/vnd.ms-excel' to the support file types.

**Review by gphorvath — APPROVED** [review]

**Pull request merged**